sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -264,8 +284,10 @@ class SparkSession private(
       Some(sharedState),
       Some(sessionState),
       extensions,
-      Map.empty)
+      Map.empty,
+      managedJobTags.asScala.toMap)
```

Comment:
qq, does this produce an immutable map (I think it should)? If so, then you don't have to force materialization.

Reply:
Unfortunately, it is a mutable concurrent map - and is backed by the same map (so changes propagate back to `managedJobTags`).